doc/decisions/constructor_functions_examples.md

```diff
@@ -0,0 +1,105 @@
+### Constructor Functions: Language examples
+
+This is not a decision.
+This file contains additional information for [Constructor Functions](constructor_functions.md).
+
+## Rust
+
+For example, `libelektra-rust` (targeted at Rust) could provide:
```

Comment:
I don't see the connection to this decision, this API can be provided in any case?

Reply:
The connection is this line in the actual decision:

> Other libraries will provide APIs on top of these functions.

You _can_ of course always provide additional APIs. However, the point here is that all languages (including C) **will** provide additional APIs to make the very minimal core API more usable. This basically allows us to create a more restricted core API.

----

I should probably also change the Rust example to something like this:

```rust
trait KeyBuilder {
    fn new(ns: Namespace, name: &[u8]) -> Self;
    fn set_value(&mut self, value: &[u8]) -> Self;
    fn set_meta(&mut self, name: &[u8], value: &[u8]) -> Self;
    fn build(self) -> Key;
}
```

With the additional explanation: If benchmarks show that this is more efficient, none of the functions above call any core APIs and `build` directly constructs a `struct _Key` via private APIs.